src/utils/daos_hdlr.c

```diff
@@ -2923,102 +2659,95 @@ dm_connect(struct cmd_args_s *ap,
 		 */
 		} else {
 			uuid_copy(ca->dst_p_uuid, ca->src_p_uuid);
-			rc = daos_pool_connect(ca->dst_p_uuid, sysname,
-					       DAOS_PC_RW, &ca->dst_poh,
+			rc = daos_pool_connect(ca->dst_p_uuid, sysname, DAOS_PC_RW, &ca->dst_poh,
 					       NULL, NULL);
 			if (rc != 0) {
 				DH_PERROR_DER(ap, rc, "failed to connect to destination pool");
-				D_GOTO(err_src, rc);
+				D_GOTO(err, rc);
 			}
 			uuid_copy(dattr.da_puuid, ca->dst_p_uuid);
 			uuid_copy(dattr.da_cuuid, ca->dst_c_uuid);
 			dattr.da_type = ca->cont_layout;
-			dattr.da_props = props;
-			rc = duns_create_path(ca->dst_poh, path,
-					      &dattr);
+			if (ca->cont_layout != DAOS_PROP_CO_LAYOUT_POSIX)
+				dattr.da_props = props;
+			rc = duns_create_path(ca->dst_poh, path, &dattr);
 			if (rc != 0) {
-				fprintf(ap->errstream, "provide a destination "
-					"pool or UNS path of the form:\n\t"
-					"--dst </$pool> | </path/to/uns>\n");
-				D_GOTO(err_dst_root, rc = daos_errno2der(rc));
+				rc = daos_errno2der(rc);
+				DH_PERROR_DER(ap, rc, "provide a destination pool or UNS path "
+					      "of the form:\n\t --dst </$pool> | </path/to/uns>");
+				D_GOTO(err, rc);
 			}
 			uuid_copy(ca->dst_c_uuid, dattr.da_cuuid);
 		}
 		/* try to open container if this is a filesystem copy,
 		 * and if it fails try to create a destination,
 		 * then attempt to open again
 		 */
-		rc = daos_cont_open(ca->dst_poh, ca->dst_c_uuid,
-				    DAOS_COO_RW, &ca->dst_coh,
+		rc = daos_cont_open(ca->dst_poh, ca->dst_c_uuid, DAOS_COO_RW, &ca->dst_coh,
 				    dst_cont_info, NULL);
 		if (rc == -DER_NONEXIST) {
 			if (ca->cont_layout == DAOS_PROP_CO_LAYOUT_POSIX) {
-				rc = dfs_cont_create(ca->dst_poh,
-						     ca->dst_c_uuid,
-						     &attr, NULL, NULL);
+				rc = dfs_cont_create(ca->dst_poh, ca->dst_c_uuid, NULL, NULL, NULL);
 				if (rc != 0) {
-					DH_PERROR_SYS(ap, rc,
-						      "failed to create destination container");
-					D_GOTO(err_dst_root, rc = daos_errno2der(rc));
+					rc = daos_errno2der(rc);
+					DH_PERROR_DER(ap, rc, "failed to create destination "
+						      "container");
```

Comment:
Minor, but it would be better to split this after a ,
```suggestion
					DH_PERROR_DER(ap, rc,
								       "failed to create destination container");
```